picard/ui/options/general.py

```diff
@@ -76,14 +76,13 @@ def load(self):
         self.ui.ignore_file_mbids.setChecked(config.setting["ignore_file_mbids"])
         self.ui.check_for_updates.setChecked(config.setting["check_for_updates"])
         self.ui.update_level.clear()
-        index = 0
-        for key in PROGRAM_UPDATE_LEVELS:
+        for index, key in enumerate(PROGRAM_UPDATE_LEVELS):
```

Comment:
Instead of testing for current level in the loop, i suggest to do it outside, and use dict features:

```python
        self.ui.update_level.clear()
        for level, description in PROGRAM_UPDATE_LEVELS.items():
            self.ui.update_level.addItem(_(description['title']), level)
        self.ui.update_level.setCurrentIndex(self.ui.update_level.findData(config.setting["update_level"]))
```

Note: since PROGRAM_UPDATE_LEVELS is a dict, it is not sure it will be ordered as written, you'd better use an ordered dict or use sorting (the former is simpler imho).

Reply:
Switched to use an OrderedDict.  That simplifies it a bit.  I also agree that it's much cleaner doing the test outside the loop as you suggested.